### ISSUE TITLE
Release to the app store?

### ISSUE THREAD

**Issue by e-alfred** [matiasdelellis/facerecognition#1]
Hello,

this looks like an interesting app. Do you intend to put it on the Nextcloud App Store? That would be awesome.

**Comment by SlavikCA** [comment]
While it's not published,
Here is how I tried to install it manually on Ubuntu:

```
sudo apt-get update composer
sudo apt-get install git-core
cd nextcloud/apps/
git clone https://github.com/matiasdelellis/facerecognition.git
cd facerecognition/
make
```

then go to Apps and enable it.

Hoping, it will get published soon...

**Comment by stalker314314** [comment]
Yap, this is about it (and also pdlib should be installed for everything to work perfectly - which is big one on itself). We should be having support for pdlib debian packages (at least).

However, do note that current state of the things is such that app is unstable, buggy, unoptimized and we probably don't think a lot about backward compatibility, so expect that you might need to purge yout database later:)

**Comment by skjnldsv** [comment]
Any news @matiasdelellis ?
This looks like an awesome app :D

**Comment by matiasdelellis** [comment]
Hi @SlavikCA 
I hope soon... :smiley: 

Although it is functional, and we have a good feedback from a few users, it is an complicated application due to its dependencies and higher requirements, and I'm worried about generating higher expectations than we can offer now. :sweat_smile:

I'm happy with the current state, and there are few details to be polished for an initial release. 
Personally:
 * I am still deciding whether to continue making debian/fedora packages for dlib and pdlib, or just automate and document the process. 
 * The biggest problem reported is always due use old dlib packages from distros and bad pdlib installation.
 * Many users expect to use CUDA that could never try CUDA due lack of hardware..
 * Therefore, we must document well, and write a good issue template to minimize these errors..
 * Well, maybe also I could wait for this... https://github.com/matiasdelellis/facerecognition/issues/163

Of course, if you did tests, I would like to know your comments..

Ragards,
Matias

**Comment by skjnldsv** [comment]
That's great news!! I just found this app to be honest! It's looking great! I did not test it yet, but I'll definitely try and experiment a bit! 
Publishing have a double effect, people expect of course more things, as the users grow, but it also gives you exposure and will attract developers and pull requests ;)

Make sure you have an inclusive and attractive readme like we try to do in the nextcloud apps, and you should be golden!

Don't hesitate if you need anything! But that many starts without being listed in the appstore is quite an accomplishment and definitely show the interest in this app!

**Comment by matiasdelellis** [comment]
Hi @skjnldsv 

I will try to publish an initial version in the next few days.

> Make sure you have an inclusive and attractive readme like we try to do in the nextcloud apps, and you should be golden!

Do you want to help us with this? Both @stalker314314 and me are not native English, so we do our best. Beyond readme, surely there are many words to improve.

**Comment by matiasdelellis** [comment]
https://github.com/nextcloud/app-certificate-requests/pull/274

:wink:

**Comment by matiasdelellis** [comment]
Hi all,
Please can suggest and test. :smile: 

**TODO:**
 - [X] New certificate.
 - [X] Update translations (Until now ..)
 - [X] Fix makefile to generate tarball.
 - [x] Deleting orphaned persons from create cluster task #179
 - [ ] Write a more friendly description and summary.. @stalker314314 
 - [ ] Update screenshots. @matiasdelellis 
 - [ ] With what version number should we publish?
 - [ ] Add an initial Changelog.
 - [ ] Issue template.. #150 @matiasdelellis 
 - [ ] ~~Decide if we publish an experimental or final version to all users. (This changes the priority of the next items)~~ It no have sense if they cannot be installed. https://github.com/nextcloud/server/issues/7417
 - [ ] Anything else you can think..? :wink: 

**Current tarball:**
 * https://services.delellis.com.ar/data/facerecognition/facerecognition.tar.gz

**Comment by skjnldsv** [comment]
> * [ ]  With what version number should we publish?

is it ready for production? Still in beta?

**Comment by matiasdelellis** [comment]
> is it ready for production? Still in beta?

Maybe stable in terms of Nexcloud release channels.. I would like to publish a beta version to request tests, but I don't see how. (The only thing would be to publish for NC18, but I don't test it yet)
So, correct me, but we must publish it, and can add a notice/warning in the summary..

**Comment by skjnldsv** [comment]
Ideally that is why we have the nightly function.
Sure as you mentioned, it's not shown in the app store (from nextcloud, https://github.com/nextcloud/server/issues/7417. Otherwise it IS shown in apps.nextcloud.com) 
I would say, publish this as a nightly and let's to a bit of publicity for this so you can attract devs and users :)

**Comment by matiasdelellis** [comment]
> Ideally that is why we have the nightly function.
> Sure as you mentioned, it's not shown in the app store (from nextcloud, [nextcloud/server#7417](https://github.com/nextcloud/server/issues/7417). Otherwise it IS shown in apps.nextcloud.com)
> I would say, publish this as a nightly and let's to a bit of publicity for this so you can attract devs and users? :)

ohh.. You're right, we can publish an initial version only there.. and wait for feedback since we have no hurry. :wink:

**Comment by matiasdelellis** [comment]
Hi all,
![Screenshot_2019-11-21 Settings - Nube](https://user-images.githubusercontent.com/733715/69332831-40041c80-0c36-11ea-9e18-24d180706b5f.png)

I went around the screenshots a lot... First I did with personal data but better not... :see_no_evil:  Then with Blinspot photos, but I assumed that the words facial recognition plus FBI sounds a bit scary... :sweat_smile: 
Then I came up with Big Bang Tehory. What do you think?

**Comment by stalker314314** [comment]
Awesome! These are real, not photoshopped, right?:)

**Comment by matiasdelellis** [comment]
> Awesome! These are real, not photoshopped, right?:)

Yes. 194 photos downloaded from Google.. :smile: 
Then i will search more and we can use them for testing. :wink:

**Comment by stalker314314** [comment]
Ultimate tests would be with Chad Smith and will Ferrell:) Joking aside, this is really good showcase and picked example!

**Comment by matiasdelellis** [comment]
> Ultimate tests would be with Chad Smith and will Ferrell:) Joking aside, this is really good showcase and picked example!

:+1: 
It is a very good example for testing the stability clusters, etc. But for screenshots is better show some consistency. :grimacing: 
PR #183

**Comment by matiasdelellis** [comment]
@stalker314314 
I'm seeing if I forget something, but I think we are at a good point to publish it. Correct me if you want to do something else. Of course there is no urgency in doing this.

The idea is to publish it for a few days (or as long as necessary) as a nightly build. So, it would be published in https://apps.nextcloud.com/, but it cannot be installed directly from nextcloud.

About the points that leave pending:
* Version: There is no reason to change it.. So remains as v0.5.8
* Initial Changelog: Just a summary of the current features. Automatic clustering, search by name of person in files, side panel where to observe and rename people. I do not think it is necessary to expand so much there..

@skjnldsv
Any comments or recommendations to make? Something that I'm forgetting?. 

@e-alfred @SlavikCA Something to tell us? :smiley:

**Comment by skjnldsv** [comment]
> * Version: There is no reason to change it.. So remains as v0.5.8

and when ready, bump it to 1.0.0 ;)

> Any comments or recommendations to make? Something that I'm forgetting?.

It's a bit troublesome to build the external dlib if you're having nextcloud in a docker container :)
Would be awesome to have steps for this use case ;)

**Comment by matiasdelellis** [comment]
> >  Version: There is no reason to change it.. So remains as v0.5.8
> 
> and when ready, bump it to 1.0.0 ;)

There is a long way to go before saying 1.0.. :wink: 

> > Any comments or recommendations to make? Something that I'm forgetting?.
> 
> It's a bit troublesome to build the external dlib if you're having nextcloud in a docker container :)
> Would be awesome to have steps for this use case ;)

It's still ugly, but I think @agboom has something functional in https://github.com/matiasdelellis/facerecognition/issues/175#issuecomment-554786383
And @Happyfeet01 could also make it work, and promised (No pressure.. :sweat_smile: ) to make a tutorial..

I have little experience in docker, I confess that its use for Nexcloud seems ridiculous until recently, but knowing Alpine changes my perspectives. :sweat_smile: 

Can you help us with that? However we follow it in #160

**Comment by matiasdelellis** [comment]
D'Oh!.
![imagen](https://user-images.githubusercontent.com/733715/69485010-c415fa00-0e18-11ea-9d22-22b07d67bd6c.png)

It seems that there is a limit of 20 Mb.. :disappointed:

**Comment by Happyfeet01** [comment]
> > > Version: There is no reason to change it.. So remains as v0.5.8
> > 
> > 
> > and when ready, bump it to 1.0.0 ;)
> 
> There is a long way to go before saying 1.0.. 😉
> 
> > > Any comments or recommendations to make? Something that I'm forgetting?.
> > 
> > 
> > It's a bit troublesome to build the external dlib if you're having nextcloud in a docker container :)
> > Would be awesome to have steps for this use case ;)
> 
> It's still ugly, but I think @agboom has something functional in [#175 (comment)](https://github.com/matiasdelellis/facerecognition/issues/175#issuecomment-554786383)
> And @Happyfeet01 could also make it work, and promised (No pressure.. 😅 ) to make a tutorial..
> 
> I have little experience in docker, I confess that its use for Nexcloud seems ridiculous until recently, but knowing Alpine changes my perspectives. 😅
> 
> Can you help us with that? However we follow it in #160

Yes I would try to make an tutorial for Docker. The Setup without Docker works really fine.

**Comment by skjnldsv** [comment]
> Of course it seems an acceptable limit, but our application weighs 27.9Mb

Wow, what is weighting that much? :O

**Comment by stalker314314** [comment]
> Wow, what is weighting that much? :O

@skjnldsv Well, neural network models for face recognition:/ @matiasdelellis is now trying to postpone downloading after installation in https://github.com/matiasdelellis/facerecognition/pull/189. Needless to say, this is going to affect already rough installation to be slightly more cumbersome:) We are also wondering where we can get those models stored after installation, because they will get overwritten on app upgrade if we keep them in app folder. What are our options, do you might know?

**Comment by skjnldsv** [comment]
> What are our options, do you might know?

Does this neural network model plan to increase its size in the future?
We could bump the max size to 30MB?

**Comment by matiasdelellis** [comment]
Hi @skjnldsv 

> Does this neural network model plan to increase its size in the future?
> We could bump the max size to 30MB?

In principle, we can meet a limit like that, but I don't know in the future.

Although today we use a group of fixed models, there are prospects of allowing supporting several models... so although I wanted to avoid it, making them installable by the administrator was a logical step.

Well, since https://github.com/matiasdelellis/facerecognition/pull/189 (Merged) after installing dlib, PDlib, and this application, you must run `occ face:setup`, to install the models we use now.

**Comment by matiasdelellis** [comment]
p.s: These are installed in `nextcloud/appdata_INSTANSE_ID/facerecognition/models` simil like previews, richdocuments templates, deck images, cms_pico, etc.. Do you think it's OK?

@stalker314314 We publish?

**Comment by e-alfred** [comment]
@matiasdelellis Did you already try to increase the compression level using Gzip? If the size of the models grow over time, maybe an additional "Install models" button in the settings menu that downloads the models manually could be a workaround?

**Comment by matiasdelellis** [comment]
@e-alfred 
> If the size of the models grow over time, maybe an additional "Install models" button in the settings menu that downloads the models manually could be a workaround?

No.. The administrator panel will only warn that the models are not installed, but their installation at least for now must be by console...

You need to access the console to run the analysis. `sudo -u apache php occ face:background_job`, otherwise it does nothing. :sweat_smile: 

Now you only have one previous step that must be executed only once.. `sudo -u apache php occ face:setup`

Beyond documenting it, it is not difficult at all. :wink:

**Comment by skjnldsv** [comment]
> p.s: These are installed in `nextcloud/appdata_INSTANSE_ID/facerecognition/models` simil like previews, richdocuments templates, deck images, cms_pico, etc.. Do you think it's OK?

depends if the process is done automatically by a script (ok) or if the admin needs to put it there (not ok)

When you say 'model' you mean the trained network data?
If so we should maybe just make it downloadable from github? Attach the model file to the release and when installing the app allow the admin to trigger the download/update? I'm not familiar with how your app works (yet :wink:) so I'll let your expertise guide me :)

**Comment by matiasdelellis** [comment]
> depends if the process is done automatically by a script (ok) or if the admin needs to put it there (not ok)

`sudo -u apache php occ face:setup` do that

> When you say 'model' you mean the trained network data?

Yes.. :wink:  

> If so we should maybe just make it downloadable from github? 

They are downloaded from Github, but from a repository that is not ours. 
 * https://github.com/davisking/dlib-models

We intend to keep it that way, because we trust them, and we don't want to suggest that we can train new models. :sweat_smile: 

> Attach the model file to the release and when installing the app allow the admin to trigger the download/update?

Just that.. `sudo -u apache php occ face:setup` but if you update the app you should not do it again. Only when installing. (Well, maybe in the future if they want to change the model.) 

> I'm not familiar with how your app works (yet wink) so I'll let your expertise guide me :)

It seems more difficult than it is .. You have to try, but meeting the requirements, everything seems to work. :slightly_smiling_face:

**Comment by stalker314314** [comment]
> 
> 
> p.s: These are installed in `nextcloud/appdata_INSTANSE_ID/facerecognition/models` simil like previews, richdocuments templates, deck images, cms_pico, etc.. Do you think it's OK?
> 
> @stalker314314 We publish?

We publish! 🙋‍♂️💪 :)

> Did you already try to increase the compression level using Gzip?

@e-alfred We could try, but it is dead end approach. Even if we fit it, next model (HOG detector) will be above limits, so better to do it in "proper" way from start, even if that means that users will have steeper onboarding:( (at least, it will be consistent throught upgrades and new versions)

**Comment by matiasdelellis** [comment]
We are close, but not yet .. :sweat_smile: 

![image](https://user-images.githubusercontent.com/733715/69552467-44ee0680-0f7d-11ea-85a7-8b6723c43df0.png)

**ERROR DETAIL**

```
database.xml did not validate:
Element 'type': [facet 'enumeration']
The value 'json_array' is not an element of the set {'integer', 'clob', 'text', 'timestamp', 'numeric', 'smallint', 'bigint', 'decimal', 'float', 'string', 'text', 'guid', 'binary', 'blob', 'boolean', 'date', 'datetime', 'datetimetz', 'time', 'dateinterval', 'json'}., line 247
```

**DATABASE DETAIL**
https://github.com/matiasdelellis/facerecognition/blob/c17fc9c3e21c7e59656ead78ebefc85141479c2b/appinfo/database.xml#L245-L258

**RESUME:**
@stalker314314 Do we switch to json or claim [appstore repo](https://github.com/nextcloud/appstore) ?. By travis we know that it works in all databases and is not reported by `occ app:check-code  facerecognition. `

**Comment by matiasdelellis** [comment]
According to doctrince 'json_array' is deprecaped in favor of json.

https://www.doctrine-project.org/projects/doctrine-dbal/en/2.9/reference/types.html#json-array

**Comment by stalker314314** [comment]
I am OK with changing it:/ Underlying schema (physical in DB) will not be changed (e.g. we don't need any upgrade step)?

**Comment by skjnldsv** [comment]
please do not use database.xml.
This is deprecated :)

Use migration paths. https://docs.nextcloud.com/server/latest/developer_manual/app/storage/migrations.html#app-db-migrations

**Comment by skjnldsv** [comment]
> We intend to keep it that way, because we trust them, and we don't want to suggest that we can train new models. 
> 
> >

perfect! :)
So why is the app 27MB then? Let's not ship it with the models and let the admin download as part of the setup process?

**Comment by matiasdelellis** [comment]
>  I am OK with changing it:/ Underlying schema (physical in DB) will not be changed (e.g. we don't need any upgrade step)?

```
[matias@nube facerecognition]$ git diff
diff --git a/appinfo/database.xml b/appinfo/database.xml
index cf73fde..274b77b 100644
--- a/appinfo/database.xml
+++ b/appinfo/database.xml
@@ -244,14 +244,14 @@
                        </field>
                        <field>
                                <name>landmarks</name>
-                               <type>json_array</type>
+                               <type>json</type>
                                <notnull>true</notnull>
                                <!-- todo: compress this once we don't need to debug this that much -->
                                <comments>JSON array that specifies values that comprise the landmarks obtained from used model</comments>
                        </field>
                        <field>
                                <name>descriptor</name>
-                               <type>json_array</type>
+                               <type>json</type>
                                <notnull>true</notnull>
                                <!-- todo: compress this once we don't need to debug this that much -->
                                <comments>JSON array of 128 value that comprise face descriptor obtained from used model</comments>
diff --git a/appinfo/info.xml b/appinfo/info.xml
index 173e190..7c7416c 100644
--- a/appinfo/info.xml
+++ b/appinfo/info.xml
@@ -16,7 +16,7 @@
 - **🚀 Build your own thing:** FaceRecognition app is just a basic building block. Through FaceRecognition API, you can build your advanced scenarios - automatically add tags to images, connect contacts and persons, share images from specific person… We want to hear your ideas!
 ]]>
        </description>
-       <version>0.5.8</version>
+       <version>0.5.9</version>
        <licence>agpl</licence>
        <author>Matias De lellis</author>
        <author>Branko Kokanovic</author>
[matias@nube facerecognition]$ sudo -u apache php ../../occ face:background_job
[sudo] password for matias: 
1/10 - Executing task CheckRequirementsTask (Check all requirements)
	Applying the memory limit to 1024MB configured in settings.
2/10 - Executing task CheckCronTask (Check that service is started from either cron or from command)
3/10 - Executing task LockTask (Acquire lock so that only one background task can run)
4/10 - Executing task DisabledUserRemovalTask (Purge all the information of a user when disable the analysis.)
5/10 - Executing task StaleImagesRemovalTask (Crawl for stale images (either missing in filesystem or under .nomedia) and remove them from DB)
6/10 - Executing task CreateClustersTask (Create new persons or update existing persons)
	Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
	* have 1000 faces already processed (you have 0),
	* have 100 images (you have 0),
	* or you need to have 95% of you images processed (you have 0.00%)
	Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
	* have 1000 faces already processed (you have 0),
	* have 100 images (you have 0),
	* or you need to have 95% of you images processed (you have 0.00%)
	Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
	* have 1000 faces already processed (you have 0),
	* have 100 images (you have 0),
	* or you need to have 95% of you images processed (you have 0.00%)
	Clusters already exist, but there was some change that requires recreating the clusters
	1800 faces found for clustering
	863 persons found after clustering
	Clusters already exist, but there was some change that requires recreating the clusters
	973 faces found for clustering
	509 persons found after clustering
	Skipping cluster creation, not enough data (yet) collected. For cluster creation, you need either one of the following:
	* have 1000 faces already processed (you have 0),
	* have 100 images (you have 0),
	* or you need to have 95% of you images processed (you have 0.00%)
7/10 - Executing task AddMissingImagesTask (Crawl for missing images for each user and insert them in DB)
	Skipping image scan for user admin that has disabled the analysis
	Skipping image scan for user educar that has disabled the analysis
	Skipping image scan for user ftoledo that has disabled the analysis
	Skipping image scan for user yany7213 that has disabled the analysis
8/10 - Executing task EnumerateImagesMissingFacesTask (Find all images which don't have faces generated for them)
9/10 - Executing task ImageProcessingTask (Process all images to extract faces)
	NOTE: Starting face recognition. If you experience random crashes after this point, please look FAQ at https://github.com/matiasdelellis/facerecognition/wiki/FAQ
10/10 - Executing task UnlockTask (Release obtained lock)
[matias@nube facerecognition]$ 


```

It seems to work just like that .. :relieved:

**Comment by matiasdelellis** [comment]
> please do not use database.xml.
> This is deprecated :)
> 
> Use migration paths. https://docs.nextcloud.com/server/latest/developer_manual/app/storage/migrations.html#app-db-migrations

Do you remember that I said we had enough to go before 1.0? :see_no_evil: 

> perfect! :)
> So why is the app 27MB then? Let's not ship it with the models and let the admin download as part of the setup process?

We wanted to simplify it a bit by sending the models in the same release. Although one more command is easy, surely there are people who will skip it and will report errors for it. :sweat_smile:

**Comment by matiasdelellis** [comment]
Finally, we have an official release as nighly.. :smile:

https://apps.nextcloud.com/apps/facerecognition

Simply, thank you very much everyone... :smiley: 

@stalker314314 I especially thank you again for all the work that you did. :smiley: :+1: :grimacing: :clap: 

:grinning:

**Comment by stalker314314** [comment]
Awesome! Congratulation! I'll have some beer tonight:)

We are not done though - my wishlist are HOG detector and support for external storage!;)

**Comment by matiasdelellis** [comment]
Very good plan... :smile:

**Comment by stalker314314** [comment]
> 
> 
> Very good plan... 😄

Yes, beer is always a good plan:p

**Comment by skjnldsv** [comment]
Congrats everyone!!!

**Comment by agboom** [comment]
That's awesome, congrats!

In an effort to make installation easier for Alpine users, I'm trying to get dlib and pdlib into the Alpine official repositories (see https://github.com/alpinelinux/aports/pull/12165 for dlib, I'm working on pdlib atm). If accepted, it will first land in the `testing` repository, later in one of the stable ones.

For people who are using the Alpine based Docker image of Nextcloud, installation would involve just running `apk add pdlib` in the container/Dockerfile, instead of building the whole thing.

**Comment by matiasdelellis** [comment]
Hi everyone,
Maybe tomorrow a new nightly release. :wink: 

**Tentative changelog:**

## [0.5.10] - 2019-12-06
### Added
- Add a button to show all clusters with same person name.
- Add a button to go back and show all clusters.

### Changed
- Select the name when open the rename dialog.
- Remove the spaces at both ends of the names before saving them.
- Force delete invalid entries when a file does not exist. Issue #154
- Improve tests broken since adding the face:setup command.
- Move from deprecated database.xml to use DB migrations.
- Try to be consistent in that we work with clusters at least in api/endpoints.
- Fix most of the 'App is not compliant' reports. Issue #72

### Translations
- Please, only one string was added. I invite you to translate them .. :wink:

**Comment by matiasdelellis** [comment]
Hi everyone,

New nightly release..
* https://github.com/matiasdelellis/facerecognition/pull/206

If there are no any comments, in a couple of hours update translations and make the publication. :wink: 

## [0.5.11] - 2019-12-20
### Added
- Add custom exclusion folder option beyond the .nomedia file. Issue #171
- Add sidebar to folders which allows to enable/disable these.
- Add support for encrypted storage. Issue #201
- Add experimental support for shared storage. Issue #26
- Add support to Nextcloud 18

### Changed
- Fix travis CI test.
- General cleaning of a lot of code and doc.

### Translations
- Update Chinese language thanks to Jack Frost.
- Update Spanish language thanks to Matias De lellis.

**Comment by matiasdelellis** [comment]
Welll... 
Finally our application was light .. hahaha
https://github.com/nextcloud/server/pull/19017

**Comment by matiasdelellis** [comment]
Hi everyone,
After so long I hope you are still interested .. :sweat_smile: 

Since I last wrote here, there has been an interesting release, and tomorrow i will publish a [new one](https://github.com/matiasdelellis/facerecognition/pull/241), which is probably the last o nighly before release a public version. :smile: 

```
[0.5.13] - 2020-03-29
Added
    Add face:reset --all|--clustering command. Details in README.md file.
    Add face:stats command. Details in README.md file.
    Add face:progress command. Details in README.md file.
    Multiple model support was implemented. Details on Wiki.
    Allow to change the model using the face:setup command.
    Allow configure the supported mimetype. Details on Wiki.

Changed
    Change to use area of the images instead of memory as the main parameter.
    Don't upsampling image on CNN, better pass a bigger picture.
    Don't install any model by default, just dump list of models.
    Better message and print system/php memory as debug.
    Test: Move to new repo that use reprepro and use bionic.

Bug fixes
    Fix show not grouped on main view.

Translations
    Add Portuguese (Brazil) thanks to Marcelo Rovani.
    Update Chinese language thanks to Jack Frost.
    Update German language thanks to Mark Ziegler.
    Update Spanish language thanks to Matias De lellis.

[0.5.12] - 2020-01-06
Bug fixes
    Force cast to integer after multiplying by a number. Issue #199, #208
    Fix initial view when enable analysis and still don't analyze anything. Issue #225 and others.
    face-preview: Person photo is blank #226

Changed
    Change the minimum memory requirement to 2GB. If you have less use Swap.
    Change the minimum memory assigned for analysis to 1.2GB.
    Change the maximum memory assigned for analysis to 8 GB.
    Change the formula to calculate the area according to memory. Issue $220, #176, and others.
    Implement a settings service where to handle everything a little more clean.
    Move FaceManagementService together with the others services.
    Test node binary needed to build handlebars templates. Issue #223 and #217
    Fix some grammatical errors and typos. Issue 224

Translations
    Add German translation thanks to Mark Ziegler.
    Update Chinese language thanks to Jack Frost.
    Update Spanish language thanks to Matias De lellis.
```

In particular, there was a major refactoring in the last days, that improve a lot of the code, and leaves us better prepared to improve the application in the future. :smile: 

I will only wait for the tests to publish this latest version, but I would like those who are interested, please try it, report errors, suggest ideas, if they can translate, so that the next version is well polished to officially publish in the app store. :smiley: 

Thanks,
Matias

**Comment by matiasdelellis** [comment]
Hello everyone,
Very probably tomorrow we will publish the first version officially in the Nextcloud appstore. :tada:

It took us two years from this issue? Wow.. It is a long time.. :sweat_smile: and let's say that with this version we are just starting... :smile: 

This is the changelog: 
## [0.6.0] - 2020-05-04
### Added
- Experimental support of External Storage. Issue #212
- Add some documentation about how expect to install, test and use the application
- Optionally you can install Pdlib 1.0.2 from https://github.com/matiasdelellis/pdlib that increase the speed of clustering drastically.
- The sidebar has been rewritten to show it in the Photos application.
- Add php-bzip2 as a necessary dependency to install the models.

### Changed
- Also adds a minimum php memory requirement for HOG. 128 MB.
. Indicate which model is enabled in the summary table of the setup --model command.

### Bug fixes
- Consider the memory configured in php as a dependency on the models.
- Improves the description of errors and prints the links to documentation whenever it can.
- Fix plurals in admin panel. Issue #256.
- Use paginated queries when search persons on file app. Fix part of issue #263.
- Search mainly faces, rather than persons in file view. Fix issue #264.
- Don't compare a good face with a bad one. This greatly improves the quality of the faces clusters.
- Fix that sidebar says no faces were found when just not clustered yet. Issue #255.
- Fix tests on Nextcloud 18 and 19.

### Translations
- New French tranlation thanks to Florian Carpentier
- Update Chinese translation thanks to yui874397188
- Update German translation thanks to Johannes Szeibert
- Update Polish translation thanks to Piotr Esse
- Update Spanish translation thanks to Matias De lellis

Thank you very much to all.. . but as always one more request..
Anyone who can test, report, translate or just comment is more than welcome. :smiley:

**Comment by matiasdelellis** [comment]
Hi everyone,

I just publish the app officially in the Nextcloud App Store.

* https://apps.nextcloud.com/apps/facerecognition

I didn't want to stop thanking everyone. Especially @stalker314314 for much of the work, but many thanks to you who test, report bugs, or just supported the project with good words. :smiley: 

It was never a project with great ambitions, but I guess it adds a feature desired by many people. Hopefully moderately meet expectations.
Certainly troubling to others, but we take the trouble that each user decides how to use it, and even which folders will be analyzed, trying to respect the privacy of all.

As I said several times, some features are missing, surely there will be errors,but we think we are in a good state to officially publish it. We could say that we reached the point where it meets our expectations, and we need feedback from real users to continue.

Thank you very much again,
I do not have much more to say. :sweat_smile: 

Just that, I hope you test it, report errors, translate, and hopefully I hope I don't disappoint you.. :smiling_face_with_three_hearts:

**Comment by stalker314314** [comment]
Cheers mate! Good luck going forward and let me wish for v1.0 no more than next year:D

**Comment by agboom** [comment]
This is great, congrats on the release :tada:! I just installed it from the app store, it works like a charm. Thanks for all the hard work!

The `face:setup` command needed to be run again to detect new photos, though. But after that the background task was happily running again :smile:

**Comment by matiasdelellis** [comment]
Hi all,
Small survey to know what are the main errors of the application.

* https://delellis.com.ar/apps/forms/LFQZz5AJrASaHqqz

Basically I try to know how reliable the clustering is, knowing the amount of images and the configured parameters.

Any other comments, as always welcome.
